lib/dvb/pmt.cpp

```diff
@@ -715,16 +726,8 @@ int eDVBServicePMTHandler::getProgramInfo(program &program)
 			program.defaultAudioStream = autoaudio_ddp;
 		else
 		{
-			if (autoaudio_mpeg != -1)
-				program.defaultAudioStream = autoaudio_mpeg;
-			else if (autoaudio_ac3 != -1)
-				program.defaultAudioStream = autoaudio_ac3;
-			else if (autoaudio_ddp != -1)
-				program.defaultAudioStream = autoaudio_ddp;
-			else if (autoaudio_aache != -1)
-				program.defaultAudioStream = autoaudio_aache;
-			else if (autoaudio_aac != -1)
-				program.defaultAudioStream = autoaudio_aac;
+			if (autoaudio != NULL)
+				program.defaultAudioStream = *autoaudio;
 			else if (first_non_mpeg != -1 && (defaultac3 || defaultddp))
```

Comment:
using && (defaultac3 || defaultddp) is from my point of view not the best solution in this part of code where everything else (use cache, use ac3, use ac3+, find stream based on languge) failed.
I would suggest adding another option, for example:
default_non_mpeg: Use non mpeg track when no set audio language was found.